test/linalg/tridiag.jl

```diff
@@ -228,118 +16,180 @@ end
+            @test istril(Tridiagonal(dl,d,zeros(du)))
+        end
 
-            A = SymTridiagonal(a, b)
+        @testset for mat_type in (Tridiagonal, SymTridiagonal)
+            A = mat_type == Tridiagonal ? mat_type(dl, d, du) : mat_type(d, dl)
             fA = map(elty <: Complex ? Complex128 : Float64, Array(A))
-
+            @testset "similar, size, and copy!" begin
+                B = similar(A)
+                @test size(B) == size(A)
+                if mat_type == Tridiagonal # doesn't work for SymTridiagonal yet
+                    copy!(B, A)
+                    @test B == A
+                end
+                @test isa(similar(A), mat_type{elty})
+                @test isa(similar(A, Int), mat_type{Int})
+                @test isa(similar(A, Int, (3, 2)), Matrix{Int})
+                @test size(A, 3) == 1
+                @test size(A, 1) == n
+                @test size(A) == (n, n)
+                @test_throws ArgumentError size(A, 0)
+            end
             @testset "getindex" begin
-                @test_throws BoundsError A[n+1,1]
-                @test_throws BoundsError A[1,n+1]
-                @test A[1,n] == convert(elty,0.0)
-                @test A[1,1] == a[1]
+                @test_throws BoundsError A[n + 1, 1]
+                @test_throws BoundsError A[1, n + 1]
+                @test A[1, n] == convert(elty, 0.0)
+                @test A[1, 1] == d[1]
             end
             @testset "setindex!" begin
                 @test_throws BoundsError A[n + 1, 1] = 0 # test bounds check
                 @test_throws BoundsError A[1, n + 1] = 0 # test bounds check
-                @test ((A[3, 3] = A[3, 3]) == A[3, 3]; A == fA) # test assignment on the main diagonal
-                @test_throws ArgumentError A[3, 2] = 1 # test assignment on the subdiagonal
-                @test_throws ArgumentError A[2, 3] = 1 # test assignment on the superdiagonal
-                @test_throws ArgumentError A[1, 3] = 1 # test assignment off the main/sub/super diagonal
+                @test_throws ArgumentError A[1, 3]   = 1 # test assignment off the main/sub/super diagonal
+                if mat_type == Tridiagonal
+                    @test (A[3, 3] = A[3, 3]; A == fA) # test assignment on the main diagonal
+                    @test (A[3, 2] = A[3, 2]; A == fA) # test assignment on the subdiagonal
+                    @test (A[2, 3] = A[2, 3]; A == fA) # test assignment on the superdiagonal
+                    @test ((A[1, 3] = 0) == 0; A == fA) # test zero assignment off the main/sub/super diagonal
+                else
+                    @test ((A[3, 3] = A[3, 3]) == A[3, 3]; A == fA) # test assignment on the main diagonal
+                    @test_throws ArgumentError A[3, 2] = 1 # test assignment on the subdiagonal
+                    @test_throws ArgumentError A[2, 3] = 1 # test assignment on the superdiagonal
+                end
             end
             @testset "Diagonal extraction" begin
```

Comment:
Could also be worth adding a test for `diag(A)`, i.e. without specifying a diagonal.